### PR TITLE
Add pub/sub channel for pool hashrate and invalid shares

### PR DESCRIPTION
**Pull request by sven-hash**

### CODE REVIEW

File: lib/pool.js [hunk 1]
```diff
@@ -209,44 +236,59 @@ var pool = module.exports = function pool(config, logger){
                     client.previousDifficulty,
                     client.difficulty,
                     client.remoteAddress,
-                    client.socket.localPort
+                    client.socket.localPort,
+                    jobExpiryPeriod
```
Comment (Lbqds) [inline lib/pool.js:240]: We don't need this parameter anymore。

**Review by Lbqds — APPROVED** [review]
One minor comment, otherwise LGTM, thanks.

**Comment by alex-oleshkevich** [comment]
bump, any updates on it?

**Comment by sven-hash** [comment]
Correct me if I'm wrong @Lbqds but these change are already merge

I think we can close this PR

**Comment by alex-oleshkevich** [comment]
Not merged. I can see only log string.
https://github.com/alephium/mining-pool/blob/master/lib/pool.js#L76

**Comment by Lbqds** [comment]
No, it's not merged. I'm not very sure if every pool needs this.

**Comment by sven-hash** [comment]
I will let you close it if you think it's not for public pool purpose

**Comment by alex-oleshkevich** [comment]
@Lbqds, a frontend that wants to display invalid shares stats will need it. if this is not in the mainstream pool, then people have to maintain their own forks just to publish invalid shares to redis.

**Comment by Lbqds** [comment]
> @Lbqds, a frontend that wants to display invalid shares stats will need it. if this is not in the mainstream pool, then people have to maintain their own forks just to publish invalid shares to redis.

The purpose of this PR seems to be to limit the number of logs through a rate limiter, it doesn't publish all valid/invalid shares to redis, do I understand this right? @sven-hash

**Comment by sven-hash** [comment]
It always publish in the log files but it's useful to "listen" instead of polling the log files

**Pull request closed without merging**